### PR TITLE
Update LUA defintion links

### PR DESCRIPTION
**Pull request by projecteon**

### CODE REVIEW

File: lua/README.md [hunk 1]
```diff
@@ -81,8 +84,8 @@ The first thing to do when writing a lua script for MQ2 is to `require('mq')`. T
 `print('string')`
 :   as noted before
 
-`mq.delay(val, callback)`
-:   where val can be an integer (which denotes milliseconds of delay) or a string that ends in 's' 'm' or 'ms' to have delays with human readable durations and callback is an optional lua function which evaluates to true or false to decide whether to end the delay early
+`mq.delay(val, --[[optional]]callback)`
+:   where val can be an integer (which denotes milliseconds of delay) or a string that ends in 's' 'm' or 'ms' to have delays with human readable durations. The callback is optinal and is a function call condition that returns a bool that can end the delay early.
```
Comment (Knightly1) [inline lua/README.md]: "optional" but I also prefer the wording "which evaluates to true or false" over "call condition"

File: lua/macro-to-lua.md [hunk 2]
```diff
@@ -14,7 +14,9 @@ Before getting started, its recommended to have [VS Code](https://code.visualstu
 
 ## Resources
 - [Programming in Lua](https://www.lua.org/pil/)
-- [Coldblooded's Lua autocompletion](https://gitlab.com/Coldblooded/mq-emmylua-definitions)
+- Coldblooded's Lua autocompletion. Requires [this](https://github.com/sumneko/lua-language-server) VSCode extension.
+    - [MQ definitions](https://github.com/johnfking/mq-definitions)
+    - [IMGUI definitions](https://github.com/johnfking/mq-imgui-definitions)
```
Comment (Knightly1) [inline lua/macro-to-lua.md]: I would prefer not to have this duplicated, it creates a maintenance issue if anything changes.  You can have it See Also for the reference to the other page which would get them to the same place.

Note this goes for both the PIL and the line here.

File: reference/data-types/datatype-type.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Contains information about data types
 | **Type** | **Member** | **Description** |
 | :--- | :--- | :--- |
 | [_string_](datatype-string.md) | **Name** | Type name |
-| [_string_](datatype-string.md) | **TypeMember**[_N_] | Member name based on an internal ID number (based on 1 through _N_, not all values will be used) |
-| [_int_](datatype-int.md) | **TypeMember**[_name_] | Member internal ID number based on _name_ (will be a number from 1 to N) |
+| [_string_](datatype-string.md) | **Member**[_N_] | Member name based on an internal ID number (based on 1 through _N_, not all values will be used) |
```
Comment (Knightly1) [inline reference/data-types/datatype-type.md:14]: Is this correct?  I didn't look at source.

Comment (projecteon) [reply reference/data-types/datatype-type.md:14]:
```
/echo ${Type[spawn].TypeMember[ID]}
No such 'type' member 'TypeMember'

/echo ${Type[spawn].Member[ID]}
1

/echo ${Type[spawn].Member[1]}
ID
```
https://docs.macroquest.org/reference/top-level-objects/tlo-type/

**Pull request merged**